### PR TITLE
osmo gamm e2e test

### PR DESCRIPTION
**Pull request by bekauz**
work towards #93 

introduces a cross chain workflow operated from main domain.

along with the tests, a few changes are made to the way manager handles polytone logic (wrt. instantiate2 address prediction, and remote domain handling of callbacks).

### CODE REVIEW

File: program-manager/src/program_config.rs [hunk 1]
```diff
@@ -212,8 +232,8 @@ impl ProgramConfig {
 
                 // Add processor bridge account info by domain
                 self.authorization_data
-                    .set_processor_bridge_addr(domain.clone(), processor_bridge_account_addr);
-            };
+                    .set_processor_bridge_addr(processor_bridge_account_addr);
+            }
```
Comment (keyleu) [inline program-manager/src/program_config.rs:236]: do you not need to know the domain of that processor bridge account? How is this change not breaking anything?

File: local-interchaintest/src/utils/authorization.rs [hunk 2]
```diff
@@ -459,3 +461,75 @@ pub fn set_up_external_domain_with_polytone(
...
+        let query_processor_callbacks_response: Value = serde_json::from_value(
+            contract_query(
+                test_ctx
+                    .get_request_builder()
+                    .get_request_builder(authorization_domain),
+                authorization_addr,
+                &serde_json::to_string(
+                    &valence_authorization_utils::msg::QueryMsg::ProcessorCallbacks {
+                        start_after: None,
+                        limit: None,
```
[the remark this answers is not in the document]
Comment (bekauz) [reply local-interchaintest/src/utils/authorization.rs:485]: ah that's good to know. happy to add it if others think it's worth it at this moment?

[the remark this answers is not in the document]
Comment (keyleu) [reply local-interchaintest/src/utils/authorization.rs:485]: nah no need to change, not worth it for testing purposes

**Review by keyleu — APPROVED** [review]

**Review by stiiifff — APPROVED** [review]

**Comment by stiiifff** [comment]
LGTM 🚀

**Pull request merged**